app/applications/features/applications-table/applications-table.hooks.tsx

```diff
@@ -0,0 +1,137 @@
+    queryParams: {
+      applicantId: githubUserId,
+    },
+    options: { enabled: Boolean(githubUserId) },
+  });
+
+  const applications = useMemo(() => data?.pages?.flatMap(data => data.applications) ?? [], [data]);
+  const hasApplications = Boolean(applications.length);
+
+  function handleProjectClick(slug: string) {
+    openProjectOverview({ slug });
+  }
+
+  function handleOpenDrawer({ issueId, applicationId }: { issueId: number; applicationId: string }) {
+    setApplyIssueDrawerState(prevState => ({ ...prevState, isOpen: true, issueId, applicationId }));
+  }
+
+  const columns: TTable.Column[] = useMemo(
+    () => [
+      {
+        key: "CREATED_AT",
+        children: <Translate token={"v2.pages.applications.table.columns.date"} />,
+        icon: {
+          remixName: "ri-time-line",
+        },
+        width: 120,
+      },
+      {
+        key: "project",
+        children: <Translate token={"v2.pages.applications.table.columns.project"} />,
+        icon: {
+          remixName: "ri-folder-3-line",
+        },
+      },
+      {
+        key: "repository",
+        children: <Translate token={"v2.pages.applications.table.columns.repository"} />,
+        icon: {
+          remixName: "ri-git-repository-line",
+        },
+      },
+      {
+        key: "contribution",
+        children: <Translate token={"v2.pages.applications.table.columns.contribution"} />,
+        icon: {
+          remixName: "ri-stack-line",
+        },
+      },
+      {
+        key: "actions",
+        children: "",
+        align: "end",
+        width: 150,
+      },
+    ],
+    []
+  );
+
+  const rows: TTable.Row[] = useMemo(
+    () =>
+      applications.map(row => {
+        const repoName = row.issue.repo.name;
+        const truncateLength = 200;
+        const contribution = mapIssueToContribution({
+          ...row.issue,
+          author: { ...row.issue.author, isRegistered: false },
+          repository: { ...row.issue.repo, owner: "" },
+          createdAt: row.receivedAt,
+        });
+
+        return {
+          key: String(row.id ?? ""),
+          CREATED_AT: <div className={"whitespace-nowrap"}>{displayRelativeDate(row.receivedAt)}</div>,
+          project: (
+            <Avatar.Labelled avatarProps={{ src: row.project.logoUrl, shape: "square" }}>
+              <Link.Button
+                onClick={() => handleProjectClick(row.project.slug)}
+                className="whitespace-normal text-left"
+                title={row.project.name}
+              >
+                {row.project.name}
+              </Link.Button>
+            </Avatar.Labelled>
+          ),
+          repository: (
+            <Link href={row.issue.repo.htmlUrl} className="whitespace-nowrap text-left" title={repoName}>
+              {truncateLength && repoName.length > truncateLength
+                ? repoName.substring(0, truncateLength) + "..."
+                : repoName}
```

Comment:
put that in a constant ? 
```suggestion
              {repoName.length > truncateLength
                ? repoName.substring(0, truncateLength) + "..."
                : repoName}
```